File: Week-02/day-04/student-filter.py
students = [
        {'name': 'Rezso', 'age': 9.5, 'candies': 2},
        {'name': 'Gerzson', 'age': 10, 'candies': 1},
        {'name': 'Aurel', 'age': 7, 'candies': 3},
        {'name': 'Zsombor', 'age': 12, 'candies': 5}
]

# create a function that takes a list of students and prints:
# - Who has got more candies than 4 candies

# create a function that takes a list of students and prints:
#  - how many candies they have on average

def candies_filter(list):
    list_max_candies = []
    for i in range(0, len(list)):
        for key, value in list[i].items():
            if key == 'candies':
                if value > 4:
                    list_max_candies.append(list[i])
    for i in range(0, len(list_max_candies)):
        for key, value in list_max_candies[i].items():
            if key == 'name':
                print(value)

candies_filter(students)

def candies_average(list):
    sum_candies = 0
    for i in range(0, len(list)):
        for key, value in list[i].items():
            if key == 'candies':
                sum_candies += value
    print(sum_candies / len(list))

candies_average(students)

File: Week-07/Day-03/student-filter.js
'use strict';

var students = [
        {'name': 'Rezso', 'age': 9.5, 'candies': 2},
        {'name': 'Gerzson', 'age': 10, 'candies': 1},
        {'name': 'Aurel', 'age': 7, 'candies': 3},
        {'name': 'Zsombor', 'age': 12, 'candies': 5}
]

// create a function that takes a list of students and logs:
// - Who has got more candies than 4 candies

// create a function that takes a list of students and logs:
//  - how many candies they have on average

function candyMoreThan4(list) {
  var listOfOwners = [];
  list.forEach(function(el) {
    if (el.candies > 4) {
      listOfOwners.push(el.name);
    }
  })
  console.log( 'Who has more than 4 candies: ' + listOfOwners);
}

function averageCandies(list) {
  var sum = 0;
  list.forEach(function(el) {
    sum += el.candies;
  })
  console.log( sum / list.length );
}

candyMoreThan4(students);

averageCandies(students);
